common/lib/dependabot/package_manager.rb

```diff
@@ -58,18 +58,36 @@ def deprecated?
       deprecated_versions.include?(version)
     end
 
+    # Raises an error if the current package manager version is unsupported.
+    # If the version is unsupported, it raises a ToolVersionNotSupported error.
+    sig { void }
+    def raise_if_unsupported!
+      return unless unsupported?
+
+      # Example: v2.*, v3.*
+      supported_versions_message = supported_versions.map { |v| "v#{v}.*" }.join(", ")
+
+      raise ToolVersionNotSupported.new(
+        name,
+        version.to_s,
+        supported_versions_message
+      )
+    end
+
     # Checks if the current version is unsupported.
     # Returns true if the version is in the unsupported_versions array; false otherwise.
     # @example
     #   package_manager.unsupported? #=> false
     sig { returns(T::Boolean) }
     def unsupported?
```

Comment:
Just note that this method written in general however the implementation can override this so bundler v1 package manager currently implementing this.